yjit/src/codegen.rs

```diff
@@ -2880,7 +2873,14 @@ fn gen_setinstancevariable(
             asm.write_label(skip_wb);
         }
     }
-    asm.stack_pop(1); // Keep it on stack during ccall for GC
+    let val = asm.stack_pop(1); // Keep it on stack during ccall for GC
+
+    if recv_opnd != SelfOpnd {
+        asm.stack_pop(1); // Pop receiver
+
+        let out_opnd = asm.stack_push(Type::Unknown); // Push a return value
+        asm.mov(out_opnd, val);
+    }
```

Comment:
This confused me for a bit, as in "wait, why are we not producing an output value if the operand is self? why are we producing a value if the operand is not self?"

Probably deserves a comment.

Reply:
wrote comments to clarify that https://github.com/ruby/ruby/pull/9986/commits/2b73548263dd9acd903f52f68cfc6ba566d0a5bf